app/services/spree_shopify_importer/data_savers/zones/zone_base.rb

```diff
@@ -0,0 +1,74 @@
+module SpreeShopifyImporter
+  module DataSavers
+    module Zones
+      class ZoneBase < BaseDataSaver
+        delegate :attributes, to: :parser
+
+        private
+
+        def assign_spree_zone_to_data_feed
+          @shopify_data_feed.update!(spree_object: @spree_zone)
+        end
+
+        def parser
+          @parser ||= SpreeShopifyImporter::DataParsers::Zones::BaseData.new(shopify_shipping_zone)
+        end
+
+        def shopify_shipping_zone
+          @shopify_shipping_zone ||= ShopifyAPI::ShippingZone.new(data_feed)
+        end
+
+        def create_spree_zone_members
+          if attributes[:kind] == 'country'
+            find_countries_members
+          else
+            find_states_members
+          end
+        end
+
+        def find_countries_members
+          @shopify_shipping_zone.countries.map do |country|
+            country_id = Spree::Country.find_by(iso: country.code)&.id
+            if !country_id && country.code == '*'
+              Spree::Country.find_each do |country|
+                create_country_member(country)
+              end
+            else
+              create_country_member(country)
+            end
+          end
+        end
+
+        def create_country_member(country)
+          Spree::ZoneMember.find_or_create_by(
+            zoneable_type: 'Spree::Country',
+            zoneable_id: country.id,
+            zone_id: @spree_zone.id)
+        end
+
+        def find_states_members
+          @shopify_shipping_zone.countries.each do |country|
+            country_id = Spree::Country.find_by(iso: country.code).id
```

Comment:
as I see above the situation when `find_by` can return nil is possible, may be here you also need safe operator before `id`?